### PR TITLE
make sure to raise an error if `pick_python_cmd` returns False for Python bundles/packages

### PR DESCRIPTION
**Pull request by casparvl**
This was already done for PythonPackage, but not for PythonBundle. I ran into this here https://github.com/easybuilders/easybuild-easyconfigs/pull/21307#issuecomment-2327000584

### CODE REVIEW

**Comment by casparvl** [comment]
When used with `ReFrame-4.9.2.eb` from https://github.com/easybuilders/easybuild-easyconfigs/pull/21307 , with this PR, a proper error is raised:
```
== 2024-09-03 21:20:47,564 run.py:247 INFO running cmd: /usr/bin/python3 -c 'import sys; print("%s.%s.%s" % sys.version_info[:3])'
== 2024-09-03 21:20:47,591 run.py:689 DEBUG cmd "/usr/bin/python3 -c 'import sys; print("%s.%s.%s" % sys.version_info[:3])'" exited with exit code 0 and output:
3.6.8

== 2024-09-03 21:20:47,591 run.py:725 DEBUG Using default regular expression: (?<![(,-]|\w)(?:error|segmentation fault|failed)(?![(,-]|\.?\w)
== 2024-09-03 21:20:47,591 pythonpackage.py:130 DEBUG Minimal requirement for minor Python version not satisfied: 3.6.8 vs 3.7
== 2024-09-03 21:20:47,591 pythonpackage.py:159 DEBUG Python command '/usr/bin/python3' does not satisfy version requirements (maj: 3, min: 7), moving on
== 2024-09-03 21:20:47,611 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:126 in __init__): Failed to pick Python command to use (at easybuild/easyblocks/generic/pythonbundle.py:115 in prepare_step)
== 2024-09-03 21:20:47,612 build_log.py:267 INFO ... (took < 1 sec)
== 2024-09-03 21:20:47,615 config.py:700 DEBUG software install path as specified by 'installpath' and 'subdir_software': /home/it4i-casparl/.local/easybuild/software
== 2024-09-03 21:20:47,615 filetools.py:2013 INFO Removing lock /home/it4i-casparl/.local/easybuild/software/.locks/_home_it4i-casparl_.local_easybuild_software_ReFrame_4.6.2.lock...
== 2024-09-03 21:20:47,616 filetools.py:383 INFO Path /home/it4i-casparl/.local/easybuild/software/.locks/_home_it4i-casparl_.
local_easybuild_software_ReFrame_4.6.2.lock successfully removed.
== 2024-09-03 21:20:47,616 filetools.py:2017 INFO Lock removed: /home/it4i-casparl/.local/easybuild/software/.locks/_home_it4i-casparl_.local_easybuild_software_ReFrame_4.6.2.lock
== 2024-09-03 21:20:47,616 easyblock.py:4285 WARNING build failed (first 300 chars): Failed to pick Python command to use
```
Although it is still cryptic (it doesn't specifically complain about the version _unless_ you run with `--debug`), at least behavior is now the same as for PythonPackage. And: at least EasyBuild doesn't just try to run a python command that doesn't exist on the system.

**Comment by casparvl** [comment]
Test build in https://github.com/easybuilders/easybuild-easyconfigs/pull/21307#issuecomment-2327320046 (ended up in the other PR... but still shows it in the gist all right: https://gist.github.com/casparvl/07e98bf3cb8c43ee1a6f34a38b1eed6d . It now

1) errors out also for `pythonbundle` and
2) the error is more clear on the fact that there is a requirement for the python versions in the EasyConfigs (and that _that_ might be why no _valid_ python command is found)

File: easybuild/easyblocks/generic/pythonbundle.py [hunk 1]
```diff
@@ -109,8 +109,20 @@ def prepare_step(self, *args, **kwargs):
 
             python_cmd = pick_python_cmd(req_maj_ver=req_py_majver, req_min_ver=req_py_minver)
 
-        self.all_pylibdirs = get_pylibdirs(python_cmd=python_cmd)
-        self.pylibdir = self.all_pylibdirs[0]
+        if python_cmd:
+            self.log.info("Python command being used: %s", python_cmd)
+        else:
+            if req_py_majver is not None or req_py_minver is not None:
```
[the remark this answers is not in the document]
Comment (bedroge) [reply easybuild/easyblocks/generic/pythonbundle.py:115]: Indeed, when testing it with for instance `ReFrame-4.6.2-GCCcore-13.2.0.eb`, I get:
```
    if (req_py_majver is not None or req_py_minver is not None or
       ^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'req_py_majver' where it is not associated with a value
```

[the remark this answers is not in the document]
Comment (casparvl) [reply easybuild/easyblocks/generic/pythonbundle.py:117]: Should be fixed by https://github.com/easybuilders/easybuild-easyblocks/pull/3430/commits/8b47e7be2f37480a84795810e9a77d23f762678e

**Comment by bedroge** [comment]
Test report by @bedroge
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _ReFrame-4.6.2.eb_ (partial log available at https://gist.github.com/bedroge/d99ddfdd92ae3a5beac6b7d85835ef41)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen3), Python 3.6.8
See https://gist.github.com/bedroge/658cf22fd1c22abb32ca8337a5027899 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _pyBigWig-0.3.22-gfbf-2023a.eb_ 
 * **SUCCESS** _PyYAML-6.0-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _sktime-0.25.0-gfbf-2023a.eb_ 
 * **SUCCESS** _NiBabel-4.0.2-foss-2022a.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 24.04.1 LTS (Noble Numbat), x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.12.3
See https://gist.github.com/bedroge/390dbacfe8775eb02397ca838813083c for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _ReFrame-4.6.2.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 24.04.1 LTS (Noble Numbat), x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.12.3
See https://gist.github.com/bedroge/ddf841dd1329ad5b5c6f7ac561446ff9 for a full test report.

**Review by bedroge — APPROVED** [review]
Lgtm

**Pull request merged**